### ISSUE TITLE
"IllegalArgumentException: LocalVariablesSorter only accepts expanded frames" when logging in PanacheRepository

### ISSUE THREAD

**Issue by bard-kristian** [quarkusio/quarkus#30173]
### Describe the bug

Using Quarkus logging in a PanacheRepository crashes the build. I'm using Kotlin, haven't tried in with Java.

### How to Reproduce?

1. Start a new (Kotlin) Quarkus project, for the reproducer I did `quarkus create app org.acme:rest-kotlin-quickstart --extension='kotlin,resteasy-reactive-jackson'`
2. Add these extensions `quarkus-jdbc-postgresql` and `quarkus-hibernate-orm-panache-kotlin`
3. Create a class extending PanacheRepository
4. Create a data class with @Entity annotation for the repository type since PanacheRepository is generic
5. Create a function that tries to log using e.g. `Log.error("Test Quarkus logging")` or `Log.info("Test Quarkus logging")`
6. Try to build and observe the failure.

See attached reproducer.

[rest-kotlin-quickstart.zip](https://github.com/quarkusio/quarkus/files/10345223/rest-kotlin-quickstart.zip)

### Output of `uname -a` or `ver`

Linux pop-os 5.19.16-76051916-generic #202210150742\~1666053244\~22.04\~cf07008 SMP PREEMPT_DYNAMIC Tue O x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 17.0.4.1 2022-08-12 OpenJDK Runtime Environment Temurin-17.0.4.1+1 (build 17.0.4.1+1) OpenJDK 64-Bit Server VM Temurin-17.0.4.1+1 (build 17.0.4.1+1, mixed mode, sharing)

### Quarkus version or git rev

Reproducer: 2.15.1.Final, project where I noticed the bug: 2.14.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Reproducer: Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63), project where I noticed the bug: Gradle 7.5.1

### Additional information

Stacktrace: 
```
2023-01-04 16:06:31,479 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (vert.x-worker-thread-0) Failed to start quarkus [Error Occurred After Shutdown]: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.steps.ClassTransformingBuildStep#handleClassTransformation threw an exception: java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep.handleClassTransformation(ClassTransformingBuildStep.java:258)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at org.objectweb.asm.commons.LocalVariablesSorter.visitFrame(LocalVariablesSorter.java:188)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.emitNullCheck(KotlinPanacheClassOperationGenerationVisitor.java:300)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.nullCheckReturn(KotlinPanacheClassOperationGenerationVisitor.java:513)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.invokeOperation(KotlinPanacheClassOperationGenerationVisitor.java:452)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.generate(KotlinPanacheClassOperationGenerationVisitor.java:341)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.visitEnd(KotlinPanacheClassOperationGenerationVisitor.java:557)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:748)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:424)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep.transformClass(ClassTransformingBuildStep.java:352)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:223)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:210)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        ... 5 more

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:335)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:265)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:60)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:220)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:203)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:537)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:437)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$4.handle(VertxHttpHotReplacementSetup.java:152)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$4.handle(VertxHttpHotReplacementSetup.java:139)
        at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
        at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.steps.ClassTransformingBuildStep#handleClassTransformation threw an exception: java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep.handleClassTransformation(ClassTransformingBuildStep.java:258)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at org.objectweb.asm.commons.LocalVariablesSorter.visitFrame(LocalVariablesSorter.java:188)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.emitNullCheck(KotlinPanacheClassOperationGenerationVisitor.java:300)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.nullCheckReturn(KotlinPanacheClassOperationGenerationVisitor.java:513)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.invokeOperation(KotlinPanacheClassOperationGenerationVisitor.java:452)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.generate(KotlinPanacheClassOperationGenerationVisitor.java:341)                                                                          sitor.visitEnd(KotlinPanacheClassOperationGenerationVisitor.java:557)at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVi
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:748)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:424)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep.transformClass(ClassTransformingBuildStep.java:352)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:223)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:210)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        ... 5 more

        at io.quarkus.builder.Execution.run(Execution.java:123)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:331)
        ... 18 more
Caused by: java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:918)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep.handleClassTransformation(ClassTransformingBuildStep.java:258)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: LocalVariablesSorter only accepts expanded frames (see ClassReader.EXPAND_FRAMES)
        at org.objectweb.asm.commons.LocalVariablesSorter.visitFrame(LocalVariablesSorter.java:188)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.emitNullCheck(KotlinPanacheClassOperationGenerationVisitor.java:300)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.nullCheckReturn(KotlinPanacheClassOperationGenerationVisitor.java:513)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.invokeOperation(KotlinPanacheClassOperationGenerationVisitor.java:452)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.generate(KotlinPanacheClassOperationGenerationVisitor.java:341)
        at io.quarkus.panache.common.deployment.visitors.KotlinPanacheClassOperationGenerationVisitor.visitEnd(KotlinPanacheClassOperationGenerationVisitor.java:557)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:748)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:424)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep.transformClass(ClassTransformingBuildStep.java:352)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:223)
        at io.quarkus.deployment.steps.ClassTransformingBuildStep$2.call(ClassTransformingBuildStep.java:210)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        ... 5 more
```

**Comment by geoand** [comment]
Also cc @Ladicek due to the Panache logging stuff

**Comment by Ladicek** [comment]
On the first sight, this looks similar to #20569, which I fixed in #20664. I'll take a look at this one (tomorrow).

**Comment by Ladicek** [comment]
The issue/PR numbers I posted in my previous comment are actually not what I wanted to mention. This actually looks similar to #22157, which I fixed in #22734. This time, it looks like the Kotlin Panache transformers break the logging feature somehow (I need to look deeper) -- it works just fine with the Java Panache layer.
